src/main/java/org/spdx/sbom/gradle/SpdxSbomPlugin.java

```diff
@@ -178,20 +180,33 @@ private void createTaskForTarget(
                       .set(
                           project.provider(
                               () ->
-                                  project.getRepositories().getAsMap().entrySet().stream()
+                                  getAllRepositories(project).entrySet().stream()
                                       .filter(e -> e.getValue() instanceof MavenArtifactRepository)
-                                      .map(Entry::getKey)
                                       .collect(
                                           Collectors.toMap(
-                                              e -> e,
+                                              Entry::getKey,
                                               e ->
-                                                  ((MavenArtifactRepository)
-                                                          project.getRepositories().getByName(e))
+                                                  ((MavenArtifactRepository) e.getValue())
                                                       .getUrl()))));
                 });
     aggregate.configure(t -> t.dependsOn(task));
   }
 
+  private Map<String, ArtifactRepository> getAllRepositories(Project project) {
+    Map<String, ArtifactRepository> projectRepositories = project.getRepositories().getAsMap();
+
+    // nasty cast because of https://github.com/gradle/gradle/issues/17295
```

Comment:
can you reference https://github.com/spdx/spdx-gradle-plugin/issues/73 in here for tracking purposed.

Reply:
Done